### ISSUE TITLE
Item selections needs two clicks on iOS

### ISSUE THREAD

**Issue by sebastijandumancic** [shentao/vue-multiselect#363]
Hi, 

I'm curious about the iOS behaviour of multiselect. If user clicks on iPhone on item in dropdown it only highlights the item, and only on second click the item is actually selected. Is that intended? 
On Android phones it highlights and selects in only one click. 

Thanks.

**Comment by shentao** [comment]
The mobile experience needs some improvements. I will look into it, thanks! :)

**Comment by sirchas** [comment]
Is there a workaround to fix this? I was thinking of clicking when selecting? Thank you if someone has had any ideas. Love this control!!

**Comment by keybrdist** [comment]
Im experiencing the same problem in the regular browser on a selectbox thats  in a small table column

**Comment by azamat-sharapov** [comment]
@shentao is the fix you pushed only for v2? If so, can we expect it to be on v1 too? Thanks.

**Comment by shentao** [comment]
@azamat-sharapov yeah, it was only pushed to v2. However, I’m not sure it actually fixed the problem. When I tested it, it sadly wasn’t deterministic, so there might have been some race conditions or something :/
If it works for you on v2, feel free to create a PR to v1. I will try to merge that and release it soon afterwards.

**Comment by alvaxet** [comment]
@shentao I'm still experiencing this issue on IOS, I'm using version 2.0.2

**Comment by jayjun** [comment]
I’ll be happy to debug this, but I can’t reproduce it using my iPhone nor iOS simulators on the official website (presumably >= 2.0.8).

If you’re facing this issue, please state your iOS version because I‘d love to get to the bottom of this.

**Comment by saulyz** [comment]
I've found that these issues have appeared because of fastClick usage. After removing fastClick, vue-multiselect@2.1.0 works as it should - no need for double click: IOS and Android devices.
Thanks for version improvements!

**Comment by jayjun** [comment]
@saulyz Cool, always worth testing with the library’s [homepage](https://vue-multiselect.js.org) to see if the reference implementation works.

**Comment by analuisamartins** [comment]
Hi. Just faced this issue, it has to do with the pseudo elements in multiselect__option—xxxxx.
Im my case / specific design I don’t need these styles and rewriting/removing these rules from the css it works as expected in mobile with one click.

Css Rules causing the issue:
.multiselect__option--highlight:after
.multiselect__option--selected:after
.multiselect__option--selected.multiselect__option--highlight:after

Problem explained here:
https://css-tricks.com/annoying-mobile-double-tap-link-issue/

**Comment by spotman** [comment]
I'm facing the same issue too and removing CSS rules fixed it. Thanks, @analuisamartins !

@shentao may I suggest to remove pseudo elements and replace them via regular tags? Doing this way webkit on iOS will work as intended.

**Comment by shentao** [comment]
That’s a great idea @spotman! Thank you.

**Comment by lhf168** [comment]
hi,author,Item selections needs two clicks on iOS,can you help me solve this problem? emergency！！！

**Comment by AstashevNikita** [comment]
whats up, im using vue-multiselect v 2.1.6

how said @analuisamartins im remove  css classes
.multiselect__option--highlight:after
.multiselect__option--selected:after
.multiselect__option--selected.multiselect__option--highlight:after

for example: 
.multiselect__option--highlight:after{
    all: unset;
}
 on chrom emulator all work fine, but when i install app on iphone, multiselect doesnt work. 
Im remove fastclick, how said @saulyz 
but this still doesnt work.

can anyone say what im doing incorrectly?
fnx

**Comment by spotman** [comment]
@AstashevNikita you need to completely remove these styles from CSS, without overwriting. I've done that by creating a copy of vue-multiselect CSS and commenting these rules. It's an ugly way but nothing else worked.

**Comment by programm1st** [comment]
And now i have another problem. 
I should use fastclick to delete delay, but for vue-multiselect i have to delete fastclick. 
With fastclick i have to use double click...
do any one know way how to use fastclick and vue-multiselect?

**Comment by davidanquetin** [comment]
I had this problem when using Multiselect with fasclick in vue, simply removed Fastclick and it works !

**Comment by Yelles** [comment]
> And now i have another problem.
> I should use fastclick to delete delay, but for vue-multiselect i have to delete fastclick.
> With fastclick i have to use double click...
> do any one know way how to use fastclick and vue-multiselect?

did you find a way to do that?

**Comment by Zh4rsiest** [comment]
> I'm facing the same issue too and removing CSS rules fixed it. Thanks, @analuisamartins !
> 
> @shentao may I suggest to remove pseudo elements and replace them via regular tags? Doing this way webkit on iOS will work as intended.

I was researching for a different issue not related to this package and removing :hover rules or only using them above mobile resolution fixes the double click issue :)

**Comment by DunnJM** [comment]
I'm facing this issue currently, is it the easiest way to rectify to manually override the css?

**Comment by DunnJM** [comment]
@analuisamartins @spotman sorry to tag you in this, but I've been unable to override the necessary rules to get this to work. When you say you created a local copy of the css and then removed the rules, how were able to ensure the component was using your local css as opposed to the css in the package? Thanks

**Comment by DunnJM** [comment]
Fixed for me by updating from Vue 2.5.x to Vue 2.6.x
